src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1110,11 +1116,36 @@ private void WriteTransitiveProjectReferences()
 
             private void WriteFrameworkReferences()
             {
+                if (_task.DisableTransitiveFrameworkReferences)
+                {
+                    return;
+                }
+
+                //  Currently there are only 2 different frameworks that we expect to
+                //  show up as framework references, so use a list here instead of a
+                //  HashSet
```

Comment:
I'd just use a HashSet, I don't think it will be much worse for 2 elements (looks like it will be backed by 3 element Slot struct array) and then we don't need to think hard about this every time we come across the O(N^2) here. Unless we've measured meaningful gain, I prefer the natural data structure that scales than one that needs a comment like this.

Reply:
Hmm, but we probably want the order to be deterministic, meaning we'd need SortedSet or a List AND a HashSet.... Less sure now.

Reply:
I implemented it the same way other similar methods work: the HashSet keeps track of items for which you've already called `WriteItem`